common/djangoapps/third_party_auth/provider.py

```diff
@@ -33,7 +33,7 @@ def _enabled_providers(cls):
             provider = OAuth2ProviderConfig.current(oauth2_slug)
             if provider.enabled_for_current_site and provider.backend_name in _PSA_OAUTH2_BACKENDS:
                 yield provider
-        if SAMLConfiguration.is_enabled(Site.objects.get_current(get_current_request())):
+        if SAMLConfiguration.is_enabled(Site.objects.get_current(get_current_request()), 'default'):
```

Comment:
What about the providers that are associated with a non-'default' `SAMLConfiguration`?

Reply:
That is a good point. I read this if statement as really asking, "is this instance of openedx set up to be a service provider with valid metadata?", but now it could be different for each idp, so it might make sense to change this to get the list of idps that have point to a valid SAMLConfig.